magit-gh-comments.el

```diff
@@ -416,27 +429,40 @@ which they came, add them to current magit-diff buffer."
 (defun magit-gh--refresh-comments ()
   "Refresh comments for the current PR."
   (interactive)
-  (let ((start-point (point))) ;; Why doesn't save-excursion work here??
-    (magit-gh--delete-comment-overlays)
-    (when (and (magit-gh--get-current-pr)
-               (derived-mode-p 'magit-diff-mode))
-      (-if-let* ((current-pr (magit-gh--get-current-pr))
-                 (comments (magit-gh--list-comments current-pr))
-                 (gh-diff (magit-gh--fetch-diff-from-github current-pr)))
-          (magit-gh--display-comments comments gh-diff)))
-    (goto-char start-point)))
+  (when (derived-mode-p 'magit-diff-mode)
+    (with-current-buffer (current-buffer)
+      (save-excursion
+        (magit-gh--delete-comment-overlays)
+        (-if-let* ((current-pr (magit-gh--get-current-pr))
+                   (comments (magit-gh--list-comments current-pr))
+                   (gh-diff (magit-gh--fetch-diff-from-github current-pr)))
+            (magit-gh--display-comments comments gh-diff))))))
 
 ;; (remove-hook 'magit-post-refresh-hook #'magit-gh--refresh-comments)
-(add-hook 'magit-post-refresh-hook #'magit-gh--refresh-comments)
+(add-hook 'magit-refresh-buffer-hook #'magit-gh--refresh-comments)
+;; (defun magit-gh--save-point ()
+;;   (setq magit-gh--current-point (point)))
+;; (defun magit-gh--restore-point ()
+;;   (message "Returning back to %s from %s after save" magit-gh--current-point (point))
+;;   (goto-char magit-gh--current-point))
+;; (add-hook 'magit-pre-refresh-hook #'magit-gh--save-point)
+;; (remove-hook 'magit-post-refresh-hook #'magit-gh--restore-point)
+;; (remove-hook 'magit-unwind-refresh-hook #'magit-gh--restore-point)
+
 
 ;; Capture and store the associated PR when the user views its diff
 ;; from the magit Pull Requests section
 (defun capture-current-pull-request ()
-  (let ((section-val (magit-section-value (magit-current-section))))
-    (destructuring-bind (user proj id) section-val
-      (setq magit-gh--current-pr (make-magit-gh-pr :owner user
-                                                   :repo-name proj
-                                                   :pr-number id)))))
+  (let ((section-val (magit-section-value (magit-current-section)))
```

Comment:
Note to self: let's do an audit on where we are using the variable `magit-gh--current-pr` and see if we can replace all references with a local variable that we properly thread through. Now that we don't need to advise someone else's function to capture this, maybe we can pass this value around instead of sticking it in the global context.